DisplaySettings/DisplaySettings.cpp

```diff
@@ -4208,14 +4209,16 @@ namespace WPEFramework {
                             if(types & dsAUDIOARCSUPPORT_eARC) {
                                 aPort.setStereoAuto(true,true);
                             }
-                            else if (types & dsAUDIOARCSUPPORT_ARC && (m_arcAudioEnabled != pEnable)) {
+                            else if (types & dsAUDIOARCSUPPORT_ARC && ((m_arcAudioEnabled != pEnable) || ( m_arcInitiationEvent == true))) {
                                 if (!DisplaySettings::_instance->requestShortAudioDescriptor()) {
                                     LOGERR("DisplaySettings::setEnableAudioPort (ARC-Auto): requestShortAudioDescriptor failed !!!\n");;
                                 }
                                 else {
                                     LOGINFO("DisplaySettings::setEnableAudioPort (ARC-Auto): requestShortAudioDescriptor successful\n");
                                 }
+                            
                             }
+                            m_arcInitiationEvent = false;
```

Comment:
Why m_arcInitiationEvent  is reset here?  Should it be cleaned up in Power state event handler & onArcTerminationEventHandler() ?

Reply:
Our dependency of this flag is only with the setEnableAudioPort invoked after the Arc initiation , Any calls to setEnableAudioPort without Arc initiation will be false alarm. The need is only when you have hotplug/standby wakeup we need the SAD request